api/spec/src/types.tsp

```diff
@@ -109,39 +132,54 @@ model Resource {
  * The unit of time for the interval.
  * One of: `day`, `week`, `month`, or `year`.
  */
-@friendlyName("IntervalUnit")
-union IntervalUnit {
-  @summary("Day")
-  day: "day",
-
-  @summary("Week")
-  week: "week",
-
-  @summary("Month")
-  month: "month",
-
-  @summary("Year")
-  year: "year",
+@friendlyName("RecurringPeriodEnum")
+enum Interval {
```

Comment:
Please create new types, this'll mess with the new plan API.

Reply:
Does the new plan api exist in any way currently? I rewrite these as they were also added to the entitlement types which have different signatures.

I suggest we could rename `RecurringPeriodEnum`  to  `Interval`. That's a breaking change in the type name, but it would only affect new versions of the SDKs, and make the naming more consistent. WDYT?